### PR TITLE
gdal: add recipe for 3.4.2

### PR DESCRIPTION
**Pull request by gillins**
Update to latest version. Also added `curl` and `geos` support to enable more functionality. 

I think I've incorporated all the recent changes to the existing recipe.

Please let me know if anything needs changing.

### CODE REVIEW

**Comment by Begasus** [comment]
This needs to be checked with opencv (soname changes), so probably opencv will need a revision bump
- [ ] opencv

**Comment by gillins** [comment]
> so probably opencv will need a revision bump

Should that be part of this PR?

**Comment by Begasus** [comment]
> Should that be part of this PR?

Doesn't have to be, you could check localy:
1. make sure you got this version for gdal installed
2. in Terminal try: `pkgman install opencv` (this installs both base and devel package), on 32bit should `opencv_x86_devel
3. if pkgman doesn't mention conflicting version for opencv_devel you are good, otherwise once this is merged you could create a new PR to revbump opencv (make sure your branch is clean before switching branches)

File: sci-libs/gdal/gdal-3.4.2.recipe [hunk 1]
```diff
@@ -235,6 +239,7 @@ BUILD()
 {
 	export LDFLAGS="-lnetwork"
 	expatDir=$portPackageLinksDir/lib~libexpat
+	touch config.rpath
```
Comment (Begasus) [inline sci-libs/gdal/gdal-3.4.2.recipe:242]: Is this needed, I thought this was fixed upstream?

**Comment by gillins** [comment]
> in Terminal try: pkgman install opencv (this installs both base and devel package),

I got the error below - I assume a version bump will be required?

```
Encountered problems:
problem 1: package opencv-4.5.1-5 requires lib:libgdal>=28.0.1, but none of the providers can be installed
  solution 1:
    - allow downgrade of gdal-3.4.2-1 to gdal-3.2.1-2
  solution 2:
    - do not install "providing opencv"
```

File: sci-libs/gdal/gdal-3.4.2.recipe (hunk 1 above)
Comment (gillins) [reply sci-libs/gdal/gdal-3.4.2.recipe:242]: Not sure, I saw it was added in https://github.com/haikuports/haikuports/commit/91a9f05086d4eea546a739fc80969ab4593d17d6. I'll remove it and see what happens.

Comment (Begasus) [reply sci-libs/gdal/gdal-3.4.2.recipe]: Yeah it was needed there, upon checking I saw mentions that this was fixed upstream. :)

**Comment by Begasus** [comment]
> I got the error below - I assume a version bump will be required?

Yes :) You could try a local build there for it to see if there are no additional changes needed for opencv (hence what I encountered before revbumping prior gdal :)

**Comment by gillins** [comment]
I'm getting a lot of errors like the one below when I try and build `opencv` locally. Not sure if I'm doing something wrong?
```
/sources/opencv-4.5.1/modules/core/src/hal_internal.cpp:176:23: error: too few arguments to function 'void sposv_(const char*, const int*, const int*, float*, const int*, float*, const int*, int*, size_t)'
  176 |                 sposv_(L, &m, &n, (float*)a, &lda, (float*)tmpB, &m, &lapackStatus);
      |                 ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by Begasus** [comment]
> I'm getting a lot of errors like the one below when I try and build `opencv` locally. Not sure if I'm doing something wrong?
> 
> ```
> /sources/opencv-4.5.1/modules/core/src/hal_internal.cpp:176:23: error: too few arguments to function 'void sposv_(const char*, const int*, const int*, float*, const int*, float*, const int*, int*, size_t)'
>   176 |                 sposv_(L, &m, &n, (float*)a, &lda, (float*)tmpB, &m, &lapackStatus);
>       |                 ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> ```

Did you try to see if the errors are known at the repository of opencv? Maybe opencv could use an update too (some changes could be different from gdal's versions), mind you I think opencv is also used by others?)

**Comment by gillins** [comment]
Turns out this is a known problem with `opencv` and recent `lapack`: https://github.com/opencv/opencv/issues/20561 https://github.com/opencv/opencv/issues/19846 - even someone mentioning this error on `haiku` here: https://github.com/opencv/opencv/issues/21116#issuecomment-980717681. 

Looks to be fixed by https://github.com/opencv/opencv/pull/21114 on 22 Dec. I suspect `lapack` got upgraded without `opencv` being rebuilt.... Time to upgrade `opencv` to the latest version (4.5.5?)

**Comment by Begasus** [comment]
I'll do a check here for both (on 32bit), can you check the new version for opencv with this version of gdal?

File: sci-libs/gdal/patches/gdal-3.4.2.patchset [hunk 2]
```diff
@@ -0,0 +1,43 @@
+From d4bd3978bd52d63c1085f1cf76a74987a609a057 Mon Sep 17 00:00:00 2001
+From: Sam Gillingham <gillingham.sam@gmail.com>
```
Comment (Begasus) [inline sci-libs/gdal/patches/gdal-3.4.2.patchset:2]: I think this could be upstreamed?

**Comment by Begasus** [comment]
Errors during build, maybe geotiff needs an update for 32bit for this one?

```
g -iquote /sources/gdal-3.4.2/ogr -iquote /sources/gdal-3.4.2/ogr/ogrsf_frmts -iquote /sources/gdal-3.4.2/gnm -iquote /sources/gdal-3.4.2/apps -DHAVE_AVX_AT_COMPILE_TIME -DHAVE_SSSE3_AT_COMPILE_TIME -DHAVE_SSE_AT_COMPILE_TIME -g -O2 -Wall -Wextra -Winit-self -Wunused-parameter -Wformat -Werror=format-security -Wno-format-nonliteral -Wlogical-op -Wshadow -Wmissing-include-dirs -Werror=vla -Wdate-time -Wnull-dereference -Wduplicated-cond -Wextra-semi -Wfloat-conversion -Wmissing-declarations -Wnon-virtual-dtor -Woverloaded-virtual -fno-operator-names -Wzero-as-null-pointer-constant -Wsuggest-override -Wimplicit-fallthrough  -DGNM_ENABLED -iquote /sources/gdal-3.4.2/port -I/packages/netcdf_x86-4.7.2-2/.self/develop/headers/x86  -DGDAL_COMPILATION -c -o ../reader_pleiades.lo reader_pleiades.cpp
geotiff.cpp: In member function 'virtual CPLErr GTiffRGBABand::IReadBlock(int, int, void*)':
geotiff.cpp:6514:20: error: invalid conversion from 'uint32_t*' {aka 'unsigned int*'} to 'uint32*' {aka 'long unsigned int*'} [-fpermissive]
 6514 |                    reinterpret_cast<uint32_t *>(m_poGDS->m_pabyBlockBuf),
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                    |
      |                    uint32_t* {aka unsigned int*}
In file included from gtiff.h:39,
                 from geotiff.cpp:31:
/boot/system/develop/headers/x86/tiffio.h:439:55: note:   initializing argument 4 of 'int TIFFReadRGBATileExt(TIFF*, uint32, uint32, uint32*, int)'
  439 | extern int TIFFReadRGBATileExt(TIFF*, uint32, uint32, uint32 *, int stop_on_error );
      |                                                       ^~~~~~~~
geotiff.cpp:6541:20: error: invalid conversion from 'uint32_t*' {aka 'unsigned int*'} to 'uint32*' {aka 'long unsigned int*'} [-fpermissive]
 6541 |                    reinterpret_cast<uint32_t *>(m_poGDS->m_pabyBlockBuf),
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                    |
      |                    uint32_t* {aka unsigned int*}
In file included from gtiff.h:39,
                 from geotiff.cpp:31:
```

File: sci-libs/gdal/gdal-3.4.2.recipe [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+	lib:libpng16$secondaryArchSuffix
+	lib:libproj$secondaryArchSuffix
+	lib:libssl$secondaryArchSuffix
+	lib:libtiff$secondaryArchSuffix
+	lib:libwebp$secondaryArchSuffix
+	lib:libxml2$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	lib:libzstd$secondaryArchSuffix
+	lib:libgeos_c$secondaryArchSuffix
+	lib:libcurl$secondaryArchSuffix
```
Comment (Begasus) [inline sci-libs/gdal/gdal-3.4.2.recipe:191]: Sort alphabetically (also in BUILD_REQUIRES)

Comment (gillins) [reply sci-libs/gdal/gdal-3.4.2.recipe]: done

**Comment by gillins** [comment]
> Errors during build, maybe geotiff needs an update for 32bit for this one

Looks like it is trying to build the internal copy of `geotiff` is it picking up external `geotiff` at the end of the configure? Could be some mismatch between haiku `libtiff` and the internal `geotiff`...

File: sci-libs/gdal/patches/gdal-3.4.2.patchset (hunk 2 above)
Comment (gillins) [reply sci-libs/gdal/patches/gdal-3.4.2.patchset:2]: Indeed - I'll submit a PR

**Comment by gillins** [comment]
Got a bit further with opencv 4.5.5, but now getting the following errors. Maybe something else needs to be bumped also?

```
[ 45%] Building CXX object modules/dnn/CMakeFiles/opencv_dnn.dir/src/ocl4dnn/src/ocl4dnn_lrn.cpp.o
In file included from /sources/opencv-4.5.5/modules/videoio/src/cap_ffmpeg.cpp:50:
/sources/opencv-4.5.5/modules/videoio/src/cap_ffmpeg_impl.hpp:542:5: error: 'AVBSFContext' does not name a type; did you mean 'AVIOContext'?
  542 |     AVBSFContext* bsfc;
      |     ^~~~~~~~~~~~
      |     AVIOContext
/sources/opencv-4.5.5/modules/videoio/src/cap_ffmpeg_impl.hpp: In member function 'void CvCapture_FFMPEG::init()':
/sources/opencv-4.5.5/modules/videoio/src/cap_ffmpeg_impl.hpp:589:5: error: 'bsfc' was not declared in this scope
  589 |     bsfc = NULL;
      |     ^~~~
/sources/opencv-4.5.5/modules/videoio/src/cap_ffmpeg_impl.hpp: In member function 'void CvCapture_FFMPEG::close()':
/sources/opencv-4.5.5/modules/videoio/src/cap_ffmpeg_impl.hpp:620:34: error: 'AVStream' {aka 'struct AVStream'} has no member named 'codec'
  620 |         avcodec_close( video_st->codec );
      |                                  ^~~~~
```

**Comment by Begasus** [comment]
> Got a bit further with opencv 4.5.5, but now getting the following errors. Maybe something else needs to be bumped also?
> 
> ```
> [ 45%] Building CXX object modules/dnn/CMakeFiles/opencv_dnn.dir/src/ocl4dnn/src/ocl4dnn_lrn.cpp.o
> In file included from /sources/opencv-4.5.5/modules/videoio/src/cap_ffmpeg.cpp:50:
> /sources/opencv-4.5.5/modules/videoio/src/cap_ffmpeg_impl.hpp:542:5: error: 'AVBSFContext' does not name a type; did you mean 'AVIOContext'?
>   542 |     AVBSFContext* bsfc;
>       |     ^~~~~~~~~~~~
>       |     AVIOContext
> /sources/opencv-4.5.5/modules/videoio/src/cap_ffmpeg_impl.hpp: In member function 'void CvCapture_FFMPEG::init()':
> /sources/opencv-4.5.5/modules/videoio/src/cap_ffmpeg_impl.hpp:589:5: error: 'bsfc' was not declared in this scope
>   589 |     bsfc = NULL;
>       |     ^~~~
> /sources/opencv-4.5.5/modules/videoio/src/cap_ffmpeg_impl.hpp: In member function 'void CvCapture_FFMPEG::close()':
> /sources/opencv-4.5.5/modules/videoio/src/cap_ffmpeg_impl.hpp:620:34: error: 'AVStream' {aka 'struct AVStream'} has no member named 'codec'
>   620 |         avcodec_close( video_st->codec );
>       |                                  ^~~~~
> ```

Maybe this could fix it, it's related to ffmpeg (we got ffmpeg5 now): https://git.alpinelinux.org/aports/tree/community/opencv/ffmpeg5.patch

**Comment by Begasus** [comment]
> > Errors during build, maybe geotiff needs an update for 32bit for this one
> 
> Looks like it is trying to build the internal copy of `geotiff` is it picking up external `geotiff` at the end of the configure? Could be some mismatch between haiku `libtiff` and the internal `geotiff`...

Tried with some suggestions I found over repology with no avail, even with the subfolders removed in the mentioned directory it still needs geotiff.cpp in the source ...

**Comment by gillins** [comment]
> Tried with some suggestions I found over repology with no avail, even with the subfolders removed in the mentioned directory it still needs geotiff.cpp in the source ...

OK will have a closer look. BTW when you say 32 bit do you mean the x86_gcc2 build, or ARM? (sorry new to Haiku)

**Comment by Begasus** [comment]
> > Tried with some suggestions I found over repology with no avail, even with the subfolders removed in the mentioned directory it still needs geotiff.cpp in the source ...
> 
> OK will have a closer look. BTW when you say 32 bit do you mean the x86_gcc2 build, or ARM? (sorry new to Haiku)

32bit as in x86_gcc2 yes

**Comment by gillins** [comment]
> Maybe this could fix it, it's related to ffmpeg (we got ffmpeg5 now): https://git.alpinelinux.org/aports/tree/community/opencv/ffmpeg5.patch

That worked! Good find. Some warnings below, but I can sort these out.

```
Warning: POLICY ERROR: no matching provides "lib:libopencv_barcode" for "lib/libopencv_barcode.so"
Warning: POLICY ERROR: no matching provides "lib:libopencv_barcode" for "lib/libopencv_barcode.so.4.5.5"
Warning: POLICY ERROR: no matching provides "lib:libopencv_barcode" for "lib/libopencv_barcode.so.405"
Warning: POLICY ERROR: no matching provides "lib:libopencv_wechat_qrcode" for "lib/libopencv_wechat_qrcode.so"
Warning: POLICY ERROR: no matching provides "lib:libopencv_wechat_qrcode" for "lib/libopencv_wechat_qrcode.so.4.5.5"
Warning: POLICY ERROR: no matching provides "lib:libopencv_wechat_qrcode" for "lib/libopencv_wechat_qrcode.so.405"
Warning: POLICY ERROR: "/packaging/opencv/lib/libopencv_wechat_qrcode.so" needs library "libiconv.so.2", but the package doesn't seem to declare that as a requirement
Warning: POLICY ERROR: "/packaging/opencv/lib/libopencv_wechat_qrcode.so.4.5.5" needs library "libiconv.so.2", but the package doesn't seem to declare that as a requirement
Warning: POLICY ERROR: "/packaging/opencv/lib/libopencv_wechat_qrcode.so.405" needs library "libiconv.so.2", but the package doesn't seem to declare that as a requirement
```

**Comment by gillins** [comment]
@Begasus is this good to merge? My patch got accepted upstream. I can submit an `opencv` PR once this is finished.

**Comment by Begasus** [comment]
> @Begasus is this good to merge? My patch got accepted upstream. I can submit an `opencv` PR once this is finished.

As far as I know this is still broken on x86_gcc2h, will see if I can try a new build later

**Comment by gillins** [comment]
gcc2 is not supported by GDAL (already excluded in https://github.com/haikuports/haikuports/blob/master/sci-libs/gdal/gdal-3.2.1.recipe#L125) and the build hints for GDAL specify that C++11 is required (https://gdal.org/build_hints.html#build-requirements)...

**Comment by Begasus** [comment]
> gcc2 is not supported by GDAL

x86_gcc2h also provides gcc11 (hence the secondary architecture) :)

**Comment by gillins** [comment]
Ah I just assumed that was a typo...!

**Comment by gillins** [comment]
What size is `int` with x86_gcc2h? Just looking at `libtiff`'s `CMakeLists.txt` line 304, seems it has decided that `sizeof(int)` != 4 and using long instead (which conflicts with GDAL's code which has it as an int).

**Comment by Begasus** [comment]
> What size is `int` with x86_gcc2h?

No idea there (can't even code "hello world"), I know my way around recipes though :)

**Comment by gillins** [comment]
> No idea there (can't even code "hello world"), I know my way around recipes though :)

No problem - I'll have a look tomorrow

**Comment by gillins** [comment]
OK finally had a chance to check this on a Haiku 32bit VM. When I look at the `tiffconf.h` from the `tiff4_devel` package I see that:
```
#define TIFF_UINT32_T unsigned int
```
But when I run `cmake` myself on the `tiff` sources I get:
```
#define TIFF_UINT32_T uint32_t
```
With the later definition I suspect the above error won't have happened. Not sure why the difference between what I get and the port....

**Comment by gillins** [comment]
However, I can confirm `sizeof(uint32_t) == sizeof(unsigned int) == 4` so it should be ok to do this kind of conversion. I think adding `-fpermissive` to the compile flags should be ok (as suggested by the error). Can you try adding:
```
export CFLAGS="-fpermissive"
export CXXFLAGS="-fpermissive"
```

to the recipe? Alternatively, give me some hints on compiling for a secondary architecture - nothing I've tried seems to work...

**Comment by Begasus** [comment]
> Alternatively, give me some hints on compiling for a secondary architecture - nothing I've tried seems to work...

If you want to build for secondary architecture on 32 bit you use `hp gdal_x86`

**Comment by Begasus** [comment]
> However, I can confirm `sizeof(uint32_t) == sizeof(unsigned int) == 4` so it should be ok to do this kind of conversion. I think adding `-fpermissive` to the compile flags should be ok (as suggested by the error). Can you try adding:
> 
> ```
> export CFLAGS="-fpermissive"
> export CXXFLAGS="-fpermissive"
> ```
> 
> to the recipe?

Didn't do the trick also: `cc1: warning: command-line option '-fpermissive' is valid for C++/ObjC++ but not for C`, then same error ...

**Comment by gillins** [comment]
Yes you are right, should be only `CXXFLAGS`... Did it complete?

**Comment by Begasus** [comment]
> Yes you are right, should be only `CXXFLAGS`... Did it complete?

It went allong until the laptop got fire I think ;) (closed down without notice), so I let it alone and just re-started the build.
I had to pass `--with-unix-stdio-64=no` to configure though?

**Comment by gillins** [comment]
> (closed down without notice),

I had that a few times with Haiku 32 bit in a VM, maybe there is a kernel bug somewhere...

> I had to pass --with-unix-stdio-64=no to configure

Seems `gdal` doesn't detect Haiku's 64bit file io properly (https://github.com/OSGeo/gdal/blob/release/3.4/gdal/m4/acinclude.m4#L89). Probably best to wait until upstream has completed their `cmake` migration... With a bit of luck this problem may go away (this code in master looks like it might get it right for Haiku: https://github.com/OSGeo/gdal/blob/master/cmake/helpers/configure.cmake).

**Comment by Begasus** [comment]
Encountered another error now:
```
envidataset.cpp: In static member function 'static GDALDataset* ENVIDataset::Create(const char*, int, int, int, GDALDataType, char**)':
envidataset.cpp:2728:18: error: 'pTmp' was not declared in this scope
 2728 |                 *pTmp, "=", CSLT_STRIPLEADSPACES | CSLT_STRIPENDSPACES);
      |                  ^~~~
envidataset.cpp:2729:17: error: 'pTokens' was not declared in this scope; did you mean 'popen'?
 2729 |             if (pTokens[0] != nullptr && pTokens[1] != nullptr && pTokens[2] == nullptr)
      |                 ^~~~~~~
      |                 popen
envidataset.cpp:2731:17: error: 'poDS' was not declared in this scope
 2731 |                 poDS->SetMetadataItem(pTokens[0], pTokens[1], "ENVI");
      |                 ^~~~
envidataset.cpp:2733:24: error: 'pTokens' was not declared in this scope; did you mean 'popen'?
 2733 |             CSLDestroy(pTokens);
      |                        ^~~~~~~
      |                        popen
envidataset.cpp: At global scope:
envidataset.cpp:2739:5: error: 'poDS' does not name a type
 2739 |     poDS->ProcessStatsFile();
      |     ^~~~
envidataset.cpp:2742:30: error: 'poDS' was not declared in this scope
 2742 |     const char* pszMapInfo = poDS->m_aosHeader["map_info"];
      |                              ^~~~
envidataset.cpp:2743:5: error: expected unqualified-id before 'if'
 2743 |     if( pszMapInfo != nullptr )
      |     ^~
envidataset.cpp:2750:30: error: 'poDS' was not declared in this scope
 2750 |     const char* pszRPCInfo = poDS->m_aosHeader["rpc_info"];
      |                              ^~~~
envidataset.cpp:2751:5: error: expected unqualified-id before 'if'
 2751 |     if( !poDS->bFoundMapinfo && pszRPCInfo != nullptr )
      |     ^~
envidataset.cpp:2758:32: error: 'poDS' was not declared in this scope
 2758 |     const char* pszGeoPoints = poDS->m_aosHeader["geo_points"];
      |                                ^~~~
envidataset.cpp:2759:5: error: expected unqualified-id before 'if'
 2759 |     if( !poDS->bFoundMapinfo && pszGeoPoints != nullptr )
      |     ^~
envidataset.cpp:2765:5: error: 'poDS' does not name a type
 2765 |     poDS->SetDescription(poOpenInfo->pszFilename);
      |     ^~~~
envidataset.cpp:2766:5: error: 'poDS' does not name a type
 2766 |     poDS->TryLoadXML();
      |     ^~~~
envidataset.cpp:2769:5: error: 'poDS' does not name a type
 2769 |     poDS->oOvManager.Initialize(poDS, poOpenInfo->pszFilename);
      |     ^~~~
envidataset.cpp:2773:5: error: 'poDS' does not name a type
 2773 |     poDS->bHeaderDirty = false;
      |     ^~~~
envidataset.cpp:2775:5: error: expected unqualified-id before 'return'
 2775 |     return poDS;
      |     ^~~~~~
envidataset.cpp:2776:1: error: expected declaration before '}' token
 2776 | }
      | ^
envidataset.cpp:2778:5: error: redefinition of 'static int ENVIDataset::GetEnviType(GDALDataType)'
 2778 | int ENVIDataset::GetEnviType(GDALDataType eType)
      |     ^~~~~~~~~~~
envidataset.cpp:2635:5: note: 'static int ENVIDataset::GetEnviType(GDALDataType)' previously defined here
 2635 | int ENVIDataset::GetEnviType(GDALDataType eType)
      |     ^~~~~~~~~~~
envidataset.cpp:2827:14: error: redefinition of 'static GDALDataset* ENVIDataset::Create(const char*, int, int, int, GDALDataType, char**)'
 2827 | GDALDataset *ENVIDataset::Create( const char *pszFilename,
      |              ^~~~~~~~~~~
envidataset.cpp:2684:14: note: 'static GDALDataset* ENVIDataset::Create(const char*, int, int, int, GDALDataType, char**)' previously defined here
 2684 | GDALDataset *ENVIDataset::Create( const char *pszFilename,
      |              ^~~~~~~~~~~
make -C sdts install-obj
make[2]: Entering directory '/sources/gdal-3.4.2/frmts/sdts'
/bin/sh /sources/gdal-3.4.2/libtool --mode=compile --silent --tag=CXX g++ -iquote /sources/gdal-3.4.2/port -iquote /sources/gdal-3.4.2/gcore -iquote /sources/gdal-3.4.2/alg -iquote /sources/gdal-3.4.2/ogr -iquote /sources/gdal-3.4.2/ogr/ogrsf_frmts -iquote /sources/gdal-3.4.2/gnm -iquote /sources/gdal-3.4.2/apps -DHAVE_AVX_AT_COMPILE_TIME -DHAVE_SSSE3_AT_COMPILE_TIME -DHAVE_SSE_AT_COMPILE_TIME -fpermissive -DBUILD_WITHOUT_64BIT_OFFSET -Wall -Wextra-semi -Wmissing-declarations -Wnon-virtual-dtor -Woverloaded-virtual -fno-operator-names -Wzero-as-null-pointer-constant -Wsuggest-override -Wimplicit-fallthrough  -iquote ../iso8211  -DGNM_ENABLED -iquote /sources/gdal-3.4.2/port -I/packages/netcdf_x86-4.7.2-2/.self/develop/headers/x86  -DGDAL_COMPILATION -c -o ../o/sdtsiref.lo sdtsiref.cpp
envidataset.cpp: In static member function 'static GDALDataset* ENVIDataset::Create(const char*, int, int, int, GDALDataType, char**)':
envidataset.cpp:2736:5: warning: control reaches end of non-void function [-Wreturn-type]
 2736 |     }
      |     ^
../../GDALmake.opt:668: recipe for target '../o/envidataset.lo' failed
make[2]: *** [../o/envidataset.lo] Error 1
make[2]: Leaving directory '/sources/gdal-3.4.2/frmts/raw'
GNUmakefile:15: recipe for target 'raw-install-obj' failed
make[1]: *** [raw-install-obj] Error 2
make[1]: *** Waiting for unfinished jobs....
/bin/sh /sources/gdal-3.4.2/libtool --mode=compile --silent --tag=CXX g++ -iquote /sources/gdal-3.4.2/port -iquote /sources/gdal-3.4.2/gcore -iquote /sources/gdal-3.4.2/alg -iquote /sources/gdal-3.4.2/ogr -iquote /sources/gdal-3.4.2/ogr/ogrsf_frmts -iquote /sources/gdal-3.4.2/gnm -iquote /sources/gdal-3.4.2/apps -Weffc++ -Wold-style-cast -DHAVE_MITAB -DHAVE_AVX_AT_COMPILE_TIME -DHAVE_SSSE3_AT_COMPILE_TIME -DHAVE_SSE_AT_COMPILE_TIME -fpermissive -DBUILD_WITHOUT_64BIT_OFFSET -Wall -Wextra-semi -Wmissing-declarations -Wnon-virtual-dtor -Woverloaded-virtual -fno-operator-names -Wzero-as-null-pointer-constant -Wsuggest-override -Wimplicit-fallthrough  -iquote ogrsf_frmts -iquote ogrsf_frmts/mem -iquote ogrsf_frmts/geojson -I/sources/gdal-3.4.2/ogr/ogrsf_frmts/geojson/libjson -iquote .    -DHAVE_GEOS=1 -I/packages/geos_x86-3.10.2-1/.self//packages/geos_x86-3.10.2-1/.self/develop/headers/x86 -DGNM_ENABLED -iquote /sources/gdal-3.4.2/port -I/packages/netcdf_x86-4.7.2-2/.self/develop/headers/x86  -DGDAL_COMPILATION  -c -o ogr2gmlgeometry.lo ogr2gmlgeometry.cpp
make[3]: Leaving directory '/sources/gdal-3.4.2/ogr/ogrsf_frmts/amigocloud'
```

**Comment by gillins** [comment]
Something has gone really wrong here - looking at the source (https://github.com/OSGeo/gdal/blob/v3.4.2/gdal/frmts/raw/envidataset.cpp#L2728) those line numbers don't match. Is it possible the source file has been scrambled?

**Comment by Begasus** [comment]
Finally got around to do a new (full) build ... with success (so I think this should be OK to merge), should need your changes to opencv to check if that one builds ok with this one :+1:

**Comment by gillins** [comment]
> Finally got around to do a new (full) build ... with success 

Well done! Did `--with-unix-stdio-64=no` need to be added?

> need your changes to opencv to check if that one builds ok with this one

Done in https://github.com/haikuports/haikuports/pull/6930

**Comment by Begasus** [comment]
> > Finally got around to do a new (full) build ... with success
> 
> Well done! Did `--with-unix-stdio-64=no` need to be added?

I still had it in, maybe it's only required for 32bit?

**Comment by extrowerk** [comment]
Builds fine on 64 bit

**Comment by Begasus** [comment]
Lets merge and keep track on the buildmaster, if need be the configure option could be added for the 32bit build.

**Pull request merged**